core/src/solve.jl

```diff
@@ -34,10 +34,33 @@ struct AllocationModel
     Δt_allocation::Float64
 end
 
+@enumx EdgeType flow control
+
+"""
+Type for storing metadata of nodes in the graph.
+"""
+struct NodeMetadata
+    type::Symbol
```

Comment:
Do you think it's worthwhile to create a NodeType enum as well? Would be for another PR though, this is big enough as is :)

Reply:
Might be, but that requires more maintenance when adding new node types. I haven't found a use for it yet.